libnavigation-base/src/main/java/com/mapbox/navigation/base/route/LegWaypoint.kt

```diff
@@ -0,0 +1,79 @@
+package com.mapbox.navigation.base.route
+
+import androidx.annotation.IntDef
+import com.mapbox.geojson.Point
+
+/**
+ * The class contains information about a waypoint which breaks the route into legs (not the silent one).
+ * @param location waypoint location provided in the route request via `coordinates` parameter.
+ * @param name waypoint name provided in the route request via `waypoint_names` parameter.
+ * @param target waypoint target provided in the route request via `waypoint_targets` parameter.
+ * @param type waypoint type (see [Type]]).
+ */
+class LegWaypoint internal constructor(
```

Comment:
Which disadvantages do you see in using existing `Waypoint` type?

Reply:
Exactly what you said.
We use `Waypoint` internally (it's in internal package). And it can be of 3 types. We want to continue using this class internally for our purposes, but the LegWaypoint returns in `RouteProgress` can't be silent. I want to limit that by API to avoid confusion.